transform-plugins/src/main/java/co/cask/hydrator/plugin/Decompressor.java

```diff
@@ -0,0 +1,315 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.Schema.Field;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Decompreses the configured fields using the algorithms specified.
+ */
+@Plugin(type = "transform")
+@Name("Decompressor")
+@Description("Decompresses configured fields using the algorithms specified.")
+public final class Decompressor extends Transform<StructuredRecord, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(Decompressor.class);
+  private final Config config;
+
+  // Output Schema associated with transform output.
+  private Schema outSchema;
+
+  // Output Field name to type map
+  private Map<String, Schema.Type> outSchemaMap = Maps.newHashMap();
+
+  // Map of field to decompressor type.
+  private final Map<String, DecompressorType> deCompMap = Maps.newTreeMap();
+
+  // This is used only for tests, otherwise this is being injected by the ingestion framework.
+  public Decompressor(Config config) {
+    this.config = config;
+  }
+
+  private void parseConfiguration(String config) throws IllegalArgumentException {
+    String[] mappings = config.split(",");
+    for (String mapping : mappings) {
+      String[] params = mapping.split(":");
+
+      // If format is not right, then we throw an exception.
+      if (params.length < 2) {
+        throw new IllegalArgumentException("Configuration " + mapping + " is in-correctly formed. " +
+                                             "Format should be <fieldname>:<decompressor-type>");
+      }
+
+      String field = params[0];
+      String type = params[1].toUpperCase();
+      DecompressorType cType = DecompressorType.NONE;
+
+      switch(type) {
```

Comment:
It should be possible to use ``DecompressorType.valueOf(type)`` instead of this switch case?